src/common.speech/ServiceRecognizerBase.ts

```diff
@@ -534,50 +540,69 @@ export abstract class ServiceRecognizerBase implements IDisposable {
             });
         }
 
+
+        this.privConnectionPromise = this.retryableConnect();
+
+        // Attach an empty handler to allow the promise to run in the background while
+        // other startup events happen. It'll eventually be awaited on.
+        this.privConnectionPromise.catch(() => { });
+
+        if (this.postConnectImplOverride !== undefined) {
+            return this.postConnectImplOverride(this.privConnectionPromise);
+        }
+
+        return this.privConnectionPromise;
+    }
+
+    private async retryableConnect(): Promise<IConnection> {
+        let isUnAuthorized: boolean = false;
+
         this.privAuthFetchEventId = createNoDashGuid();
-        const sessionId: string = this.privRecognizerConfig.parameters.getProperty(PropertyId.Speech_SessionId, undefined);
+        const sessionId: string = this.privRequestSession.sessionId;
         this.privConnectionId = (sessionId !== undefined) ? sessionId : createNoDashGuid();
 
         this.privRequestSession.onPreConnectionStart(this.privAuthFetchEventId, this.privConnectionId);
+        let lastStatusCode: number = 0;
+        let lastReason: string = "";
 
-        const authPromise = isUnAuthorized ? this.privAuthentication.fetchOnExpiry(this.privAuthFetchEventId) : this.privAuthentication.fetch(this.privAuthFetchEventId);
+        const retryCap = parseInt(this.privRecognizerConfig.parameters.getProperty("SPEECH-Error-MaxRetryCount", "4"));
+        while (this.privRequestSession.numConnectionAttempts <= retryCap) {
 
-        this.privConnectionPromise = authPromise.then(async (result: AuthInfo) => {
-            await this.privRequestSession.onAuthCompleted(false);
+            // Get the auth information for the connection. This is a bit of overkill for the current API surface, but leaving the plumbing in place to be able to raise a developer-customer
+            // facing event when a connection fails to let them try and provide new auth information.
+            const authPromise = isUnAuthorized ? this.privAuthentication.fetchOnExpiry(this.privAuthFetchEventId) : this.privAuthentication.fetch(this.privAuthFetchEventId);
+            var auth: AuthInfo = await authPromise;
```

Comment:
Did not know TS allowed var in this instance.

Reply:
TS does. The linter on the other hand....